### ISSUE TITLE
Sprite top, left, bottom, right is not accurate

### ISSUE THREAD

**Issue by markjoshua12** [pvcraven/arcade#647]
## Bug Report
I was messing around making my game for the Python Discord Game Jam, and I noticed something odd when I zoom in the player sprite.

![test_3](https://user-images.githubusercontent.com/48671765/79636898-bd17cb80-81ad-11ea-8b5d-cf849e8791f7.gif)

Aside from the jittering (more on that next time), the player sprite is actually 1px lower than intended.
This could be further checked by printing the values:
```
print(f"Top: {self.player.top} | Left: {self.player.right}")
print(f"Bottom: {self.player.top} | Right: {self.player.right}")
```
Output:
```
Top: 79.0 | Left: 79.0
Bottom: 79.0 | Right: 79.0
```
(Note: all the image sizes are 16x16)

Here is another test with jumping

![test_5](https://user-images.githubusercontent.com/48671765/79636811-0f0c2180-81ad-11ea-8e8f-fc50c3d9bad8.gif)

Notice the bottom and the top of the sprite is not aligned with the tiles
![image](https://user-images.githubusercontent.com/48671765/79637391-d884d580-81b1-11ea-8c7f-47eb965c0dc2.png)
![image](https://user-images.githubusercontent.com/48671765/79637411-f6523a80-81b1-11ea-9e18-9d7f9d4295e2.png)

And here is another one using the example code (04_add_gravity.py):
![image](https://user-images.githubusercontent.com/48671765/79637511-bb043b80-81b2-11ea-9f92-a0ec083a2fd9.png)
![image](https://user-images.githubusercontent.com/48671765/79637514-c2c3e000-81b2-11ea-8ffe-830051699b2a.png)
Even here the player sprite is 0.5 pixels below (which is the `TILE_SCALING` amount in the code)

### Actual behavior:
`Sprite` right and bottom is actually (right - scale) and (bottom - scale)

### Expected behavior:
`Sprite` right and bottom should be equal to its `left + image_width` and `top - image_height`

### Steps to reproduce/example code:
Print the `top`, `right`, `bottom`, or `left` of the sprite and compare it with the intended values
Example:
```
# Added in the example code 02_draw_sprites.py
# Inside the for loop to create the ground

print(f"Width: {wall._get_width()} | Height: {wall._get_height()} ")
print(f"Top: {wall.top} | Right: {wall.right} | Bottom: {wall.bottom} | Left: {wall.left}")
```
Output:
```
Width: 64.0 | Height: 64.0 
Top: 64.0 | Right: 31.5 | Bottom: 0.5 | Left: -32.0
```
Right should be 32.0 instead and the Bottom 0.0
Setting the `left` and the `bottom` instead of `center_x` or `center_y` produces this:
```
Width: 64.0 | Height: 64.0 
Top: 95.5 | Right: 63.5 | Bottom: 32.0 | Left: 0.0
```
Should be `Top: 96` and `Right: 64`

Let me know if you can replicate it. Thanks

**Comment by pvcraven** [comment]
Trying to narrow down what the report it, the following code:
```
import arcade

wall = arcade.Sprite(":resources:images/tiles/boxCrate_double.png", 0.5)
wall.position = 0, 0

print(f"{wall.position} Width: {wall.width} | Height: {wall.height} ")
print(f"Top: {wall.top} | Right: {wall.right} | Bottom: {wall.bottom} | Left: {wall.left}")
print(f"Hit box: {wall.hit_box}")
print(f"Adjusted hit box: {wall.get_adjusted_hit_box()}")
```
...shows a height of 64, so it is expected there would be a bottom coordinate of a hitbox of -32, but instead we get -31.5.

**Comment by pvcraven** [comment]
Can you try 2.4a4 please?

**Comment by markjoshua12** [comment]
Just tried using 2.4a4, I am still experiencing the same behavior.

In the meantime, I did a temporary hack which is to manually call `set_hit_box()` and add the points based on the shape of the Texture.

While reading the source I noticed in `drawing_support.py` `calculate_points()`
`bottom_border` and `right_border` is subtracted by one, which makes sense as the `image` is zero indexed.

But these same variables is used to calculate the points.
So it looks like these values should be added by one(?) to reverse the subtraction.

Though I don't know if this would affect the geometry of the shape.
